tests/etl/test_job_ingestion.py

```diff
@@ -334,6 +334,13 @@ def test_ingest_job_with_updated_job_group(test_repository, failure_classificati
     job_type but different job_group
     """
     first_job_datum = sample_data.job_data[0]
+
+    # store original test job data so we can restore it after
```

Comment:
OMG this is horrible. :) Could you modify the sample data fixture in tests/conftest.py to not have a "session" scope? We shouldn't be reusing test data between sessions for this exact reason. I am 99% sure that is the root of our problems:

https://github.com/mozilla/treeherder/blob/master/tests/conftest.py#L66